Author: Chow Loong Jin <hyperair@debian.org>
Description: Call dbus_g_thread_init before calling any dbus functions
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1048341
Index: gconf/gconf/gconf-dbus.c
===================================================================
--- gconf.orig/gconf/gconf-dbus.c	2012-09-11 09:54:10.056848441 +0800
+++ gconf/gconf/gconf-dbus.c	2012-09-11 10:07:03.967826677 +0800
@@ -396,6 +396,13 @@
       return FALSE;
     }
 
+  /* Initialize DBus Glib for multithreading -- this fixes race conditions when
+     multi-threaded applications use gconf. Additionally, although the API
+     documentation says that dbus_g_thread_init() may only be called once, it is
+     actually really a wrapper for dbus_threads_init_default() which casn be
+     called as many times as necessary. */
+  dbus_g_thread_init ();
+
   dbus_error_init (&error);
 
   global_conn = dbus_bus_get_private (DBUS_BUS_SESSION, &error);
